slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -0,0 +1,132 @@
+    auto overlapping_submaps_trimmer_2d =
+        pose_graph_options.overlapping_submaps_trimmer_2d();
+    auto ceres_scan_matcher_options =
+        pose_graph_options.constraint_builder_options()
+            .ceres_scan_matcher_options();
+    auto trajectory_builder_2d_options =
+        slamService.mapBuilder.trajectory_builder_options_
+            .trajectory_builder_2d_options();
+
+    auto tolerance = boost::test_tools::tolerance(0.00001);
+    BOOST_TEST(pose_graph_options.optimize_every_n_nodes() ==
+               slamService.optimize_every_n_nodes);
+    BOOST_TEST(
+        trajectory_builder_2d_options.submaps_options().num_range_data() ==
+        slamService.num_range_data);
+    BOOST_TEST(trajectory_builder_2d_options.missing_data_ray_length() ==
+                   slamService.missing_data_ray_length,
+               tolerance);
+    BOOST_TEST(
+        trajectory_builder_2d_options.max_range() == slamService.max_range,
+        tolerance);
+    BOOST_TEST(
+        trajectory_builder_2d_options.min_range() == slamService.min_range,
+        tolerance);
+    if (slamService.GetActionMode() == SLAMServiceActionMode::LOCALIZING) {
+        BOOST_TEST(slamService.mapBuilder.trajectory_builder_options_
+                       .pure_localization_trimmer()
+                       .max_submaps_to_keep() ==
+                   slamService.max_submaps_to_keep);
+    } else {
+        BOOST_TEST(slamService.mapBuilder.trajectory_builder_options_
+                       .pure_localization_trimmer()
+                       .max_submaps_to_keep() == 0);
+    }
+    if (slamService.GetActionMode() == SLAMServiceActionMode::UPDATING) {
+        BOOST_TEST(overlapping_submaps_trimmer_2d.fresh_submaps_count() ==
+                   slamService.fresh_submaps_count);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_covered_area() ==
+                       slamService.min_covered_area,
+                   tolerance);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_added_submaps_count() ==
+                   slamService.min_added_submaps_count);
+    } else {
+        BOOST_TEST(overlapping_submaps_trimmer_2d.fresh_submaps_count() == 0);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_covered_area() == 0,
+                   tolerance);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_added_submaps_count() ==
+                   0);
+    }
+    BOOST_TEST(ceres_scan_matcher_options.occupied_space_weight() ==
+                   slamService.occupied_space_weight,
+               tolerance);
+    BOOST_TEST(ceres_scan_matcher_options.translation_weight() ==
+                   slamService.translation_weight,
+               tolerance);
+    BOOST_TEST(ceres_scan_matcher_options.rotation_weight() ==
+                   slamService.rotation_weight,
+               tolerance);
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_mapping) {
```

Comment:
[nit] I think this test will be more likely to catch bugs if all the parameters are set to different values.

Reply:
It would catch a typo where you set parameter a to the value of parameter b, for example.

Reply:
Ah I can see that, thanks for pointing that out